Re-implemented clustering of snapshots

RandomDefaultUser opened the pull request:
Closes #246 .

ellisja reviewed (changes requested):
Very nice. Just a few small typos that I found.

RandomDefaultUser commented on mala/common/parameters.py:
```diff
@@ -401,6 +401,24 @@ class ParametersData(ParametersBase):
+        i.e. not the entire training data is used by rather only a subset
+        which is determined by a clustering algorithm.
+
+    numbers_of_cluster : int
```
Fixed in [0098144](https://github.com/mala-project/mala/pull/249/commits/0098144914ebd23b6979d060f9f2f8928eb0be41).

RandomDefaultUser commented on mala/common/parameters.py:
```diff
@@ -412,6 +430,10 @@ def __init__(self):
+        self.use_clustering = False
+        self.numbers_of_cluster = 0
+        self.train_ratio = 0.0
+        self.sample_ration = 0.0
```
Fixed in [0098144](https://github.com/mala-project/mala/pull/249/commits/0098144914ebd23b6979d060f9f2f8928eb0be41).

RandomDefaultUser commented on mala/common/parameters.py:
```diff
@@ -401,6 +401,24 @@ class ParametersData(ParametersBase):
+        to train the encoder for the clustering.
+
+
+    sample_ration : float
```
Fixed in [0098144](https://github.com/mala-project/mala/pull/249/commits/0098144914ebd23b6979d060f9f2f8928eb0be41).

RandomDefaultUser commented:
Thanks for the feedback @ellisja ! I fixed the typos. I also deleted the files in the ml-dft-sandia folder that this merge request incorporates into the mala core code. One final question - in the implementation of this PR, only the training data set is clustered, validation and test data set are never clustered. Is that alright?

ellisja commented:
> Thanks for the feedback @ellisja ! I fixed the typos. I also deleted the files in the ml-dft-sandia folder that this merge request incorporates into the mala core code. One final question - in the implementation of this PR, only the training data set is clustered, validation and test data set are never clustered. Is that alright?

Yes, that's right. The validation and test set are only used for forward inference calculations and would not benefit as much as the training set.
